protocol/lifi/types.go

```diff
@@ -0,0 +1,81 @@
+package lifi
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+)
+
+type BigInt struct {
+	*big.Int
+}
+
+func (b *BigInt) UnmarshalJSON(data []byte) error {
+	if b.Int == nil {
+		b.Int = new(big.Int)
+	}
+
+	s := strings.Trim(string(data), "\"")
+	_, ok := b.SetString(s, 10)
+	if !ok {
+		return fmt.Errorf("failed to parse big.Int from %s", s)
+	}
+
+	return nil
+}
+
+func (b *BigInt) MarshalJSON() ([]byte, error) {
+	return []byte(b.String()), nil
```

Comment:
needs to be quoted string because of big numbers that surpasses javascript precision